src/devices/sound/discrete.cpp

```diff
@@ -204,74 +209,78 @@ inline void discrete_task::step_nodes()
 		}
 	}
 
-	/* buffer the outputs */
+	// buffer the outputs
+	for (output_buffer &outbuf : m_buffers)
+		*outbuf.ptr.load(std::memory_order_relaxed) = *outbuf.source;
+	std::atomic_thread_fence(std::memory_order_release);
 	for (output_buffer &outbuf : m_buffers)
-		*outbuf.ptr++ = *outbuf.source;
+		outbuf.ptr.store(outbuf.ptr.load(std::memory_order_relaxed) + 1, std::memory_order_relaxed);
```

Comment:
This was pretty dodgy – there’s no guarantee of when the postincrement happens relative to the store.

We can use a single release barrier here between storing the samples and updating the pointers to allow some loop unrolling, etc.